src/ProjectAudioManager.cpp

```diff
@@ -944,31 +942,30 @@ bool ProjectAudioManager::DoRecord(AudacityProject &project,
             // ISO standard would be nice, but ":" is unsafe for file name.
             nameSuffix.Replace(wxT(":"), wxT("-"));
 
-            if (newTrack.get() == first) {
+            if (newTrack == first) {
```

Comment:
I made a mistake here at 03c68afc55b2020b0e8086aa288090b4e5ea8ca2, neglecting the provisions for more than two channels (maybe in future).

Correct my mistake.  Change the condition back to
`if (baseTrackName.empty())`